pyro/poutine/handlers.py

```diff
@@ -25,71 +25,132 @@
 
 def trace(fn=None, graph_type=None, param_only=None):
     """
-    :param fn: a stochastic function (callable containing pyro primitive calls)
-    :param graph_type: string that specifies the kind of graph to construct
-    :param param_only: if true, only records params and not samples
-    :returns: stochastic function wrapped in a TraceHandler
-    :rtype: pyro.poutine.TraceHandler
+    Return a handler that records the inputs and outputs of primitive calls
+    and their dependencies.
 
-    Alias for TraceHandler constructor.
+    Consider the following Pyro program:
 
-    Given a callable that contains Pyro primitive calls, return a TraceHandler callable
-    that records the inputs and outputs to those primitive calls
-    and their dependencies.
+        >>> def model(x):
+        ...     s = pyro.param("s", torch.tensor(0.5))
+        ...     z = pyro.sample("z", dist.Normal(x, s))
+        ...     return z ** 2
+
+    We can record its execution using ``trace``
+    and use the resulting data structure to compute the log-joint probability
+    of all of the sample sites in the execution or extract all parameters.
 
-    Adds trace data structure site constructors to primitive stacks
+        >>> trace = pyro.poutine.trace(model).get_trace(0.0)
+        >>> logp = trace.log_prob_sum()
+        >>> params = [trace.nodes[name]["value"].unconstrained() for name in trace.param_nodes]
```

Comment:
Which of `["value"]` or `["value"].unconstrained()` do you think is the more common use case?

Reply:
The latter is always guaranteed to be correct, so I used it